com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -93,33 +93,23 @@ static CloudControllerClient getControllerClient(DelegateExecution context, Clou
     }
 
     public static String determineCurrentUser(VariableScope scope, StepLogger stepLogger) {
-        String userId = Authentication.getAuthenticatedUserId();
-        String previousUser = (String) scope.getVariable(Constants.VAR_USER);
-        // Determine the current user
-        if (userId != null && !userId.equals(previousUser)) {
-            stepLogger.debug(Messages.AUTHENTICATED_USER_ID, userId);
-            stepLogger.debug(Messages.PREVIOUS_USER, previousUser);
-        }
+        String userId = (String) scope.getVariable(Constants.VAR_USER);
         if (userId == null) {
-            // If the authenticated user cannot be determined,
-            // use the user saved by the previous service task
-            userId = previousUser;
-            if (userId == null) {
-                // If there is no previous user, this must be the first service task
-                // Use the process initiator in this case
-                userId = (String) scope.getVariable(Constants.PARAM_INITIATOR);
-                stepLogger.debug(Messages.PROCESS_INITIATOR, userId);
-                if (userId == null) {
-                    throw new SLException(Messages.CANT_DETERMINE_CURRENT_USER);
-                }
-            }
+            userId = getProcessInitiator(scope, stepLogger);
+            scope.setVariable(Constants.VAR_USER, userId);
         }
-        // Set the current user in the context for use by later service tasks
-        scope.setVariable(Constants.VAR_USER, userId);
-
         return userId;
     }
 
+    private static String getProcessInitiator(VariableScope scope, StepLogger stepLogger) {
+        String initiator = (String) scope.getVariable(Constants.PARAM_INITIATOR);
+        stepLogger.debug(Messages.PROCESS_INITIATOR, initiator);
+        if (initiator == null) {
+            throw new SLException(Messages.CANT_DETERMINE_CURRENT_USER);

```

Comment:
It should be a responsibility of `determineCurrentUser` to throw this exception. `getProcessInitiator` now "knows" that it's the last option of finding the current user, which means that is also "knows" how `determineCurrentUser` works. That's not good, because `determineCurrentUser` is higher in terms of abstraction level. Furthermore, methods should treat other methods as blackboxes and not rely on knowledge of their internal implementation, regardless on where they stand in the abstraction hierarchy.